src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -88,9 +87,18 @@ public void resetData(ReadOnlyAddressBook newData) {
         setTags(new HashSet<>(newData.getTagList()));
         syncMasterTagListWith(persons);
 
-        setLifeInsurances(newData.getLifeInsuranceMap());
-        syncMasterLifeInsuranceMapWithPersonList();
-        syncMasterPersonListWithLifeInsurance();
+        try {
+            setLifeInsurances(newData.getLifeInsuranceMap());
+        } catch (DuplicateInsuranceException e) {
+            assert false : "AddressBooks should not have duplicate insurances";
+        }
+
+        try {
+            syncMasterLifeInsuranceMapWith(persons);
+            syncMasterPersonListWith(lifeInsuranceMap);
+        } catch (InsuranceNotFoundException e) {
+            assert false : "AddressBooks should not contain id that doesn't match to an insurance";
+        }
     }
```

Comment:
why is assertion used here rather than throwing the exception and handling it?
i see that the original code did that as well with the person list but i think it should be changed to be able to handle the exception and fix the code, maybe have an internal function that helps remove duplicates?

Reply:
TBH I'm not confident with this either.
However, I do think using `assert` is correct because if the program runs correctly, there is No Way that the user can create a "duplicated person" or a "cannot be found insurance". Both two situations only happen when there is a bug in the program, which is on the developer side and therefore, we use `assert`.